graph/src/data/query/result.rs

```diff
@@ -21,11 +21,24 @@ pub struct QueryResult {
     pub data: Option<q::Value>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub errors: Option<Vec<QueryError>>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "serialize_data"
+    )]
+    pub extensions: Option<q::Value>,
 }
 
 impl QueryResult {
     pub fn new(data: Option<q::Value>) -> Self {
-        QueryResult { data, errors: None }
+        QueryResult {
+            data,
+            errors: None,
+            extensions: None,
+        }
+    }
+    pub fn with_extensions(mut self, extensions: q::Value) -> Self {
+        self.extensions = Some(extensions);
```

Comment:
I did you one better: I just changed the type there to `extensions: BTreeMap<q::Name, q::Value>` so that the compiler checks that we construct the right extensions.

Reply:
Perfect, I didn't want to push that far because I thought it would be inconvenient. 👍